rtcstats.js

```diff
@@ -146,262 +122,191 @@ module.exports = function(wsURL, getStatsInterval, prefixesToWrap) {
+      args = undefined;
+    } else if (args.length === 1) {
+      args = args[0];
+    }
+    trace(name, args);
+    var returnValue = native.apply(this, arguments);
+    if (returnValue !== null && typeof returnValue === 'object' && 'then' in returnValue) {
+      returnValue.then(
+        function (value) { trace(name + 'OnSuccess', inspect(value)); },
+        function (failure) { trace(name + 'OnFailure', inspect(failure)); }
+      );
+    }
+    return returnValue;
+  };
+}
+
+function traceEvent(trace, object, event, eventInspector) {
+  eventInspector = eventInspector || inspect;
+  var reportedEventName = 'on' + event;
+  object.addEventListener(event, function(e) {
+    trace(reportedEventName, eventInspector(e));
+  });
+}
+
+function traceStateChangeEvent(trace, object, stateKey) {
+  var changeEventName = stateKey.toLowerCase() + 'change';
+  var reportedEventName = 'on' + changeEventName;
+  object.addEventListener(changeEventName, function () {
+    trace(reportedEventName, object[stateKey]);
+  });
+}
+
+function tracePeerConnection(pc, options) {
+  var config = options.config || { nullConfig: true };
+  var constraints = options.constraints;
+  var trace = options.trace;
+  var getStatsInterval = options.getStatsInterval;
+
+  if (!config) {
+    config = { nullConfig: true };
+  }
+
+  config = deepCopy(config); // deepcopy
+  // don't log credentials
+  ((config && config.iceServers) || []).forEach(function(server) {
+    delete server.credential;
+  });
+
+  if (isFirefox) {
+    config.browserType = 'moz';
+  } else if (isEdge) {
+    config.browserType = 'edge';
+  } else {
+    config.browserType = 'webkit';
+  }
+
+  trace('create', config);
+
+  // TODO: do we want to log constraints here? They are chrome-proprietary.
+  // http://stackoverflow.com/questions/31003928/what-do-each-of-these-experimental-goog-rtcpeerconnectionconstraints-do
+  if (constraints) {
+    trace('constraints', constraints);
+  }
+
+  [
+    'createDataChannel', 'close',
+    'addTrack', 'removeTrack',
+    'addStream', 'removeStream',
+    'createOffer', 'createAnswer',
+    'setLocalDescription', 'setRemoteDescription',
+    'addIceCandidate'
+  ].forEach(function (method) {
+    traceMethod(trace, pc, method);
+  });
+
+  [
+    'addstream', 'removestream',
+    'track',
+    'negotiationneeded',
+    'datachannel'
+  ].forEach(function (event) {
+    traceEvent(trace, pc, event);
+  });
+
+  // Safari's RTCPeerConnectionIceEvent doesn't seem to have an appropriate constructor
+  traceEvent(trace, pc, 'icecandidate', inspectors.RTCPeerConnectionIceEvent);
+
+  [
+    'connectionState',
+    'signalingState',
+    'iceConnectionState',
+    'iceGatheringState',
+  ].forEach(function (event) {
+    traceStateChangeEvent(trace, pc, event);
+  });
+
+  // TODO: do we want one big interval and all peerconnections
+  //    queried in that or one setInterval per PC?
+  //    we have to collect results anyway so...
+  if (!isEdge) {
```

Comment:
It seems that Edge 15 is supporting `getStats()` but it requires a query parameter. We might want to explore what we can get out of that.